src/library/escape.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * escape.c - Source file for escaping capability
+ * Copyright (c) 2021 Red Hat Inc.
+ * All Rights Reserved.
+ *
+ * This software may be freely redistributed and/or modified under the
+ * terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING. If not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335, USA.
+ *
+ * Authors:
+ *   Radovan Sroka <rsroka@redhat.com>
+ */
+
+#include "escape.h"
+
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+
+#include "message.h"
+
+
+#define IS_HEX(X) (isxdigit(X) > 0 && !(islower(X) > 0))
+
+static char asciiHex2Bits(char X)
+{
+	char base = 0;
+	if (X >= '0' && X <= '9') {
+		base = '0';
+	} else if (X >= 'A' && X <= 'F') {
+		base = 'A' - 10;
+  }
+	return (X - base) & 0X00FF;
+}
+
+
+/*
+ * the function returns escaped copy of the input string
+ * returned string should be freed by caller
+ * it returns NULL in case of error
+ *
+*/
+char *escape(const char *input, int mode)
+{
+	char buffer[4096 + 1] = {0};
+	size_t pos = 0;
+
+	for (size_t i = 0 ; i < strlen(input); i++) {
+		int should_escape = 0;
```

Comment:
Could we catch the length in an unsigned variable and use that in the loop?

Reply:
It should be resolved now.